### ISSUE TITLE
Turnary Operator Executes wrong/both paths with "--transform-object-keys true"

### ISSUE THREAD

**Issue by rodhoward** [javascript-obfuscator/javascript-obfuscator#948]
Using the "?" Turnary operator a function in the "dead" branch is being executed when it shouldn't be.

## Expected Behavior
Expected only the one (correct) path to be executed. 

## Current Behavior
``` test.js
const isTrue = something => !!(something?.bob || something?.sally);
const throwsError = () => {
  throw new Error("Should not be here!");
};
const myFunction = () => {
  return isTrue() ? { my: "object", anotherParam: throwsError() } : "The only place we should be";
};

console.log(myFunction());
```
Throws error!

## Steps to Reproduce
1. save above code. "test.js"
2. npx javascript-obfuscator test.js --transform-object-keys true
3. node test-obfuscated.js

Expected string "the only place we should be" but getting exception "Should not be here!".

FYI: Without the transform-object-keys option it seems to be working fine.

## Your Environment
* Obfuscator version used:  "2.12.0"
* Node version used: v14.15.3

**Comment by sanex3339** [comment]
Hmm, this is a hard case to track.
Seems to find these cases we have to traverse up to all parent nodes of each `ObjectExpression` node until we found the host node of `ObjectExpression`. If we found `ConditionalExpression` or `LogicalExpression` node, then we have to traverse over `ObjectExpression` node and if we found `CallExpression` node, then this is our case. 

But this is too many additional operations just to cover this case.

**Comment by sanex3339** [comment]
PR:
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/951

**Comment by sanex3339** [comment]
Released as `javascript-obfuscator@2.15.6`. For more performance, i used the following logic:
> If `ObjectExpression` contains `CallExpression` or `NewExpression` nodes at any place - transformation of this `ObjectExpression` will be completely ignored
